pkg/render/certs.go

```diff
@@ -38,37 +39,33 @@ func GetSecret(ctx context.Context, client client.Client, name string, ns string
 	return secret, nil
 }
 
-// EnsureCertificateSecret ensures that the certificate in the provided
-// secrets has the expected DNS names. If no key secret is provided, a new key
-// secret is created and returned. If the key secret provided does have the
-// right DNS name, then that given key secret is returned.
-// Otherwise a new key secret is created and returned.
-func EnsureCertificateSecret(ctx context.Context, secretName string, secret *corev1.Secret, svcDNSNames ...string) (*corev1.Secret, error) {
+// EnsureCertificateSecret ensures that the certificate in the
+// secret has the expected DNS names. If no secret is provided, a new
+// secret is created and returned. If the secret does have the
+// right DNS name then the secret is returned.
+// Otherwise a new secret is created and returned.
+func EnsureCertificateSecret(ctx context.Context, secretName string, secret *corev1.Secret, keyName string, certName string, certDuration time.Duration, svcDNSNames ...string) (*corev1.Secret, error) {
 	var err error
 
 	// Create the secret if it doesn't exist.
 	if secret == nil {
-		secret, err = render.CreateOperatorTLSSecret(nil,
-			secretName, "tls.key", "tls.crt",
-			render.DefaultCertificateDuration, nil, svcDNSNames...,
+		secret, err = CreateOperatorTLSSecret(nil,
+			secretName, keyName, certName,
+			certDuration, nil, svcDNSNames...,
 		)
 		if err != nil {
 			return nil, err
 		}
 		return secret, nil
 	}
 
-	// If the cert's DNS names have changed then we need to recreate the secret.
+	// If the cert's DNS names have changed or if the cert is invalid, create
+	// a new one.
 	ok, err := secretHasExpectedDNSNames(secret, svcDNSNames)
-
-	if err != nil {
-		return nil, err
-	}
-	// DNS names on the cert do not match expected values; create a new cert.
-	if !ok {
-		return render.CreateOperatorTLSSecret(nil,
+	if !ok || err != nil {
+		return CreateOperatorTLSSecret(nil,
 			secretName, "tls.key", "tls.crt",
```

Comment:
I think you need to update this to be `keyName, certName`.

Reply:
I thought I fixed this. Fixed now